### DISCUSSION TITLE
[Kavita] Zotero Integration

### DISCUSSION THREAD

**Discussion by capelo0** [Kareadita/Kavita#3375] [Ideas]
### Idea Description

Hi @majora2007 , I'm organizing all my library in Zotero as it is super helpful to deal with the citations. 
Due to that, all my books are being organized there, also because it's easy to handle the metadata. 
I've been trying some workaround, but would it be possible to integrate it in Kavita as an External Tool (similar approach than Calibre) so I could get access to the books with the metadata already there?

Thank you for your work. 
Hope this become reality 🤞 

### Idea Category

API

### Duration of Using Kavita

1 month

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by majora2007** [comment]
There doesn't seem to be any metadata standard nor indication from their website that they write metadata into the PDF (rather into their own DB)

**Reply by capelo0** [reply]
@majora2007  and isn't it possible to read from their DB?
Does that mean that Zotero would need to build a plugin to integrate with Kavita?

**Reply by majora2007** [reply]
That isn't something I would be willing to do. Zotero could potentially have a plugin, but you're talking about 2 separate programs. Pdf doesn't have a metadata standard. I doubt Zotero would build a plugin to push metadata into Kavita. Kavita doesn't support some of the same metadata, unless it's being stored in the PDF file. 

It just doesn't seem like a natural pairing imo.

**Reply by capelo0** [reply]
Kavita could leverage Zotero API to grab the Metadata. 
I think that if Zotero would do some development around this it wouldn't be a plugin to Kavita, but a simplified version of their web app presenting the book covers.

**Reply by majora2007** [reply]
That is outside of scope for me. You are free to push Zotero to add support (unlikely), but unless there is a published (and common) standard to consume, it is extremely unlikely.

**Comment by ririnto** [comment]
I think ... a helper application would be a more useful solution.

If possible, read the Zotero API and write to the Kavita API.

Kavita already has that API.
